src/tests/functional/plugin/shared/include/single_layer_tests/reverse.hpp

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "shared_test_classes/single_layer/reverse.hpp"
+
+namespace LayerTestsDefinitions {
+
+TEST_P(ReverseLayerTest, CompareWithRefs) {
```

Comment:
All `TEST_P(ReverseLayerTest, CompareWithRefs) {` to not have duplicated symbols, please see for example [`src/tests/functional/plugin/cpu/single_layer_tests/variadic_split.cpp`](https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/cpu/single_layer_tests/variadic_split.cpp#L77)

Reply:
I think that it is right location for TEST_P() of shared_test_classes. It is CPU specific test(VariadicSplitLayerCPUTest) instance in [src/tests/functional/plugin/cpu/single_layer_tests/variadic_split.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/cpu/single_layer_tests/variadic_split.cpp#L77).

Reply:
I just provided it as example.
Defining it in header file and including it in 2 separate translation unit could lead to linker error

Reply:
in general, you may be right, but not it's a common approach in all the shared tests, you can take a look folder
src/tests/functional/plugin/shared/include/single_layer_tests
so I'm not going to change this in scope of this PR as there should be common refactoring which is out of scope now